File: public/docs/_examples/pipes/ts/app/flying-heroes.pipe.ts
// #docregion
// #docregion pure
import {Flyer} from './heroes';
import {Pipe, PipeTransform} from 'angular2/core';

@Pipe({ name: 'flyingHeroes' })
export class FlyingHeroesPipe implements PipeTransform {
  transform(allHeroes:Flyer[]) {
    // #docregion filter
    return allHeroes.filter(hero => hero.canFly);
    // #enddocregion filter
  }
}
// #enddocregion pure

/////// Identical except for the pure flag
// #docregion impure
// #docregion pipe-decorator
@Pipe({
  name: 'flyingHeroes',
  pure: false
})
// #enddocregion pipe-decorator
export class FlyingHeroesImpurePipe extends FlyingHeroesPipe {}
// #enddocregion impure

File: public/docs/_examples/pipes/dart/lib/flying_heroes_pipe.dart
// #docregion
// #docregion pure
import 'package:angular2/angular2.dart';
import 'heroes.dart';

@Pipe(name: 'flyingHeroes')
class FlyingHeroesPipe extends PipeTransform {
  // #docregion filter
  List<Hero> transform(dynamic value, [List<dynamic> args]) =>
      value.where((hero) => hero.canFly).toList();
  // #enddocregion filter
}
// #enddocregion pure

// Identical except for the pure flag
// #docregion impure, pipe-decorator
@Pipe(name: 'flyingHeroes', pure: false)
// #enddocregion pipe-decorator
class FlyingHeroesImpurePipe extends FlyingHeroesPipe {}
